test/api/transactions.js

```diff
@@ -550,6 +552,73 @@ describe('GET /api/transactions/get?id=', function () {
+			node.expect(res.body).to.have.property('transaction').that.is.an('object');
+			node.expect(res.body.transaction.id).to.equal(transactionInCheck.id);
+			node.expect(res.body.transaction.amount).to.equal(transactionInCheck.amount);
+			node.expect(res.body.transaction.fee).to.equal(transactionInCheck.fee);
+			node.expect(res.body.transaction.recipientId).to.equal(transactionInCheck.recipientId);
+			node.expect(res.body.transaction.senderId).to.equal(transactionInCheck.senderId);
+			node.expect(res.body.transaction.asset).to.eql(transactionInCheck.asset);
+			done();
+		});
+	});
+
+	it('should get transaction with asset for delegate transaction', function (done) {
+		var transactionInCheck = genesisblock.transactions.find(function (trs) {
+			return trs.id === '3634383815892709956';
+		});
+
+		var params = 'id=' + transactionInCheck.id;
+		node.get('/api/transactions/get?' + params, function (err, res) {
+			node.expect(res.body.transaction.type).to.equal(transactionTypes.DELEGATE);
+			node.expect(res.body.transaction.type).to.equal(transactionInCheck.type);
+
+			node.expect(res.body).to.have.property('success').to.be.ok;
+			node.expect(res.body).to.have.property('transaction').that.is.an('object');
+			node.expect(res.body.transaction.id).to.equal(transactionInCheck.id);
+			node.expect(res.body.transaction.amount).to.equal(transactionInCheck.amount);
+			node.expect(res.body.transaction.fee).to.equal(transactionInCheck.fee);
+			node.expect(res.body.transaction.recipientId).to.equal(transactionInCheck.recipientId);
+			node.expect(res.body.transaction.senderId).to.equal(transactionInCheck.senderId);
+			node.expect(res.body.transaction.asset.username).to.equal(transactionInCheck.asset.username);
+			done();
+		});
+	});
+
+	it('should get transaction with empty asset for transfer transaction', function (done) {
+		var transactionInCheck = genesisblock.transactions.find(function (trs) {
+			return trs.id === '1465651642158264047';
+		});
+
+		var params = 'id=' + transactionInCheck.id;
+		node.get('/api/transactions/get?' + params, function (err, res) {
+			node.expect(res.body.transaction.type).to.equal(transactionTypes.SEND);
+			node.expect(res.body.transaction.type).to.equal(transactionInCheck.type);
+
+			node.expect(res.body).to.have.property('success').to.be.ok;
+			node.expect(res.body).to.have.property('transaction').that.is.an('object');
+			node.expect(res.body.transaction.id).to.equal(transactionInCheck.id);
+			node.expect(res.body.transaction.amount).to.equal(transactionInCheck.amount);
+			node.expect(res.body.transaction.fee).to.equal(transactionInCheck.fee);
+			node.expect(res.body.transaction.recipientId).to.equal(transactionInCheck.recipientId);
+			node.expect(res.body.transaction.senderId).to.equal(transactionInCheck.senderId);
+			node.expect(res.body.transaction.type).to.equal(transactionInCheck.type);
+			node.expect(res.body.transaction.asset).to.eql({});
+			done();
+		});
+	});
```

Comment:
These tests are great. My only concern is that they do not cover **all** transaction types which can have an attached asset. My preference would be to go the extra step and cover these. More difficult I know. It might be best to test the API functionality more basically, and cover type variance at unit level.